services/order.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, User, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> None:
+    with transaction.atomic():
+        user = User.objects.get(username=username)
+        order = Order.objects.create(user=user)
+        if date:
+            date = datetime.datetime.strptime(date, "%Y-%m-%d %H:%M")
```

Comment:
Ensure that the date parsing with `strptime` correctly handles potential exceptions, such as `ValueError`, to prevent the application from crashing if an invalid date format is provided.